### PR TITLE
Add more unit-test for kubelet

### PR DESCRIPTION
**Pull request by c3y1huang**
## Why is this PR needed?

Increase unit test code coverage.
Ref: SUSE/avant-garde#744

## What does this PR do?

Tests include:
* generate kubelet root cert
* generate kubelet root cert when already exist
* disarm kubelet

Other changes:
* Refactor TestDisarmKubeletJobName to consolidate duplicated codes.
* Unify test function names in kubernetes.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

```
github.com/SUSE/skuba/internal/pkg/skuba/kubernetes/kubelet.go (6.2%)
```

### Status **AFTER** applying the patch

```
github.com/SUSE/skuba/internal/pkg/skuba/kubernetes/kubelet.go (87.5%)
```

## Docs

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: internal/pkg/skuba/kubernetes/kubelet_test.go [hunk 1]
```diff
@@ -18,40 +18,91 @@
...
 
+func TestGenerateKubeletRootCert(t *testing.T) {
+	t.Run("generate kubelet root cert", func(t *testing.T) {
+		if err := GenerateKubeletRootCert(); err != nil {
+			t.Errorf("error not expected, but error reported generating pki: %v", err)
+		}
+
+		if err := GenerateKubeletRootCert(); err != nil {
+			t.Errorf("error not expected, but error reported when pki exist: %v", err)
+		}
```
Comment (jenting) [inline internal/pkg/skuba/kubernetes/kubelet_test.go]: intended? or duplicated?

Comment (c3y1huang) [reply internal/pkg/skuba/kubernetes/kubelet_test.go]: intended. this is to test `generate kubelet root cert when already exist`.

Comment (c3y1huang) [reply internal/pkg/skuba/kubernetes/kubelet_test.go]: added comment in-code.

**Comment by jenting** [comment]
It would be good to check on the generate CA certificate have `CA:True` property.

**Pull request merged**